apache-activemq-mixin/dashboards/apache-activemq-cluster-overview.libsonnet

```diff
@@ -0,0 +1,592 @@
+      mappings: [],
+      thresholds: {
+        mode: 'absolute',
+        steps: [
+          {
+            color: 'green',
+            value: null,
+          },
+          {
+            color: '#EAB839',
+            value: 50,
+          },
+          {
+            color: 'red',
+            value: 70,
+          },
+        ],
+      },
+      unit: 'percent',
+    },
+    overrides: [],
+  },
+  options: {
+    displayMode: 'gradient',
+    minVizHeight: 10,
+    minVizWidth: 0,
+    orientation: 'horizontal',
+    reduceOptions: {
+      calcs: [
+        'lastNotNull',
+      ],
+      fields: '',
+      values: false,
+    },
+    showUnfilled: false,
+    valueMode: 'color',
+  },
+  pluginVersion: '10.2.0-59981',
+};
+
+local averageBrokerMemoryUsagePanel = {
+  datasource: promDatasource,
+  targets: [
+    prometheus.target(
+      'avg by (activemq_cluster) (activemq_memory_usage_ratio{job=~"$job", activemq_cluster=~"$activemq_cluster"})',
+      datasource=promDatasource,
+      legendFormat='{{activemq_cluster}}',
+    ),
+  ],
+  type: 'bargauge',
+  title: 'Average broker memory usage',
+  description: 'Average percentage of broker memory used across clusters.',
+  fieldConfig: {
+    defaults: {
+      color: {
+        mode: 'thresholds',
+      },
+      mappings: [],
+      thresholds: {
+        mode: 'absolute',
+        steps: [
+          {
+            color: 'green',
+            value: null,
+          },
+          {
+            color: '#EAB839',
+            value: 50,
+          },
+          {
+            color: 'red',
+            value: 70,
+          },
+        ],
+      },
+      unit: 'percent',
+    },
+    overrides: [],
+  },
+  options: {
+    displayMode: 'gradient',
+    minVizHeight: 10,
+    minVizWidth: 0,
+    orientation: 'horizontal',
+    reduceOptions: {
+      calcs: [
+        'lastNotNull',
+      ],
+      fields: '',
+      values: false,
+    },
+    showUnfilled: false,
+    valueMode: 'color',
+  },
+  pluginVersion: '10.2.0-59981',
+};
+
+{
+  grafanaDashboards+:: {
+    'apache-activemq-cluster-overview.json':
+      dashboard.new(
+        'Apache ActiveMQ cluster overview',
+        time_from='%s' % $._config.dashboardPeriod,
+        tags=($._config.dashboardTags),
+        timezone='%s' % $._config.dashboardTimezone,
+        refresh='%s' % $._config.dashboardRefresh,
+        description='',
+        uid=dashboardUid,
+      )
+
+      .addTemplates(
+        [
+          template.datasource(
+            promDatasourceName,
+            'prometheus',
+            null,
+            label='Data Source',
+            refresh='load'
+          ),
+          template.new(
+            'job',
+            promDatasource,
+            'label_values(activemq_topic_producer_count,job)',
+            label='Job',
+            refresh=2,
+            includeAll=true,
+            multi=true,
+            allValues='',
+            sort=0
+          ),
+          template.new(
```

Comment:
Also need an optional "Cluster" selector because of kubernetes support. This'll be the same for all dashboards. Look at the two mixins I mentioned earlier for examples

Reply:
good to resolve